src/edu/stanford/dlss/was/WasapiConnection.java

```diff
@@ -0,0 +1,129 @@
+package edu.stanford.dlss.was;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.FileOutputStream;
+import java.io.UnsupportedEncodingException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Iterator;
+
+import org.apache.http.StatusLine;
+import org.apache.http.cookie.Cookie;
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpResponse;
+import org.apache.http.HeaderIterator;
+import org.apache.http.NameValuePair;
+import org.apache.http.client.ClientProtocolException;
+import org.apache.http.client.CookieStore;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.HttpResponseException;
+import org.apache.http.client.ResponseHandler;
+import org.apache.http.client.entity.UrlEncodedFormEntity;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.protocol.HttpClientContext;
+import org.apache.http.impl.client.BasicCookieStore;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.message.BasicNameValuePair;
+import org.apache.http.util.EntityUtils;
+
+public class WasapiConnection {
+
+    private CloseableHttpClient wasapiClient;
+    private HttpClientContext wasapiContext;
+    private BasicCookieStore cookieStore;
+
+    public WasapiConnection(WasapiDownloaderSettings settings) throws IOException, UnsupportedEncodingException {
+        wasapiClient = initializeClient();
+        HttpPost loginRequest = createLoginRequest(settings);
+        login(loginRequest);
+        wasapiContext = HttpClientContext.create();
+        wasapiContext.setCookieStore(cookieStore);
+    }
+
+    public WasapiResponse jsonQuery(String requestURL) throws IOException {
+        HttpGet jsonRequest = new HttpGet(requestURL);
+
+        ResponseHandler<WasapiResponse> rh = new ResponseHandler<WasapiResponse>() {
+            @Override
+            public WasapiResponse handleResponse(final HttpResponse response) throws IOException {
+                StatusLine statusLine = response.getStatusLine();
+                HttpEntity entity = response.getEntity();
+
+                if(validateResponse(response.getStatusLine(), (entity == null))) {
```

Comment:
you could use `statusline` here instead of calling the method again, yes?  Or would it be better to get rid of line 54?